### DISCUSSION TITLE
Energy consumption estimate based on PID-controlled heat... is it possible?

### DISCUSSION THREAD

**Discussion by hermetheuscoffee** [artisan-roaster-scope/artisan#765] [Q&A] [answered]
First off, I'm excited about the energy estimation feature in Artisan! In looking at the documentation, the setup seems straightforward for slider-controlled heat settings... but I use PID to control my heat. Is it possible to somehow use these PID-controlled values as the input to the energy estimator? I graph the "Duty %" in the extra devices screen as shown (and I label this "PID Heat"). 
![pid_heat_extra_devices](https://user-images.githubusercontent.com/95500043/146448421-954f8143-1b6b-4afe-a789-447f9abb5423.png)

Next, if I look at Events from a prior roast, I can see that these PID values are, indeed logged: 

![pid_heat_events](https://user-images.githubusercontent.com/95500043/146448580-0ba9dd40-64bc-45c8-b0fc-694cc0037c7e.png)

However, they aren't logged as "Burner" events, so they aren't calculated correctly. Is there a way for the "Loads" tab to be able to let me select "Duty %". Or alternatively, is there a way that I could create duplicate "Burner" events for every "Duty %" event?

![energy_loads](https://user-images.githubusercontent.com/95500043/146448918-007a882c-2ec3-4747-8050-5fd2b00a6f11.png)

It's a great feature, so I'm hoping this is possible. Thanks!

**Comment by roasterdave** [comment, marked as answer]
Yes it is possible.  You will want to create an Event Quantifier to map your PID Heat curve into the burner event.  This [blog article](https://artisan-roasterscope.blogspot.com/2014/04/event-quantifiers.html) explains quantifiers.  The quantifier SV option is new in v2.4.6 and you can find a description [here](https://github.com/artisan-roaster-scope/artisan/issues/608).

**Reply by hermetheuscoffee** [reply]
Well shoot, that was easy! This makes total sense. Thanks so much, Dave!
